app/models/laboratory.rb

```diff
@@ -2,10 +2,21 @@
 
 # Laboratory: represents a lab result
 class Laboratory < ApplicationRecord
+  include ValidationHelper
   belongs_to :patient
 
   validates :result, inclusion: { in: ['positive', 'negative', 'indeterminate', 'other', nil, ''] }
 
+  validates :lab_type, on: %i[api import], inclusion: {
+    in: VALID_PATIENT_ENUMS[:lab_type],
+    message: "is not an acceptable value, acceptable values are: '#{VALID_PATIENT_ENUMS[:lab_type].reject(&:blank?).join("', '")}'"
+  }
+
+  %i[specimen_collection
+     report].each do |date_field|
+    validates date_field, on: %i[api import], date: true
+  end
+
```

Comment:
should we include validation for the `report` and `result` dates as well for consistency with the UI? this is how it's currently validated there: https://github.com/SaraAlert/SaraAlert/blob/master/app/javascript/components/laboratory/Laboratory.js#L49

Reply:
I think this sort of validation was actually added by https://github.com/SaraAlert/SaraAlert/pull/413. But then the validation was rolled back since it was causing some issues and required further testing. The validator that would handle this is actually commented out here: https://github.com/SaraAlert/SaraAlert/blob/bd8f1223578781c780cefd475127832718f26e75/app/models/laboratory.rb#L10.

Reply:
got it, thanks Nick